### ISSUE TITLE
How do the third_party folder licenses apply to the firebase project as a whole? 

### ISSUE THREAD

**Issue by lakersr8ers** [firebase/firebase-android-sdk#168]
Hi, 

Our team would like to use firebase, but our legal has some concerns regarding the ICU and SGI-STL licenses linked in the third_party folder of the firebase-android-sdk repo. What are those licenses applying to? Is it specific components? (Like firestore only) Or does it apply to all components vended by firebase? We really want to use firebase, but have to understand how those licenses apply. 

License folder: 
https://github.com/firebase/firebase-android-sdk/tree/27d084e733c33ed927529e3cdd37af8702f02850/third_party/licenses

**Comment by CRamsan** [comment]
+1 for this.
I could not find anything in particular in for ICU, the libstdc++, zlib, etc. Are these files relevant to the repo?

**Comment by samtstern** [comment]
@vkryachko @ashwinraghav can you comment on this?  I believe these licenses are the licenses that apply to our open-source dependencies and don't affect the license of the Firebase SDK source.

**Comment by lakersr8ers** [comment]
@samtstern  What in firebase is using those open source dependencies?

**Comment by samtstern** [comment]
You can see here which libraries use which license:
https://github.com/firebase/firebase-android-sdk/blob/27d084e733c33ed927529e3cdd37af8702f02850/root-project.gradle#L162

For example we use the 'flatbuffers' library which uses the Apache-2.0 license

**Comment by vkryachko** [comment]
@lakersr8ers essentially this represents licenses of our transitive dependencies that it is not possible to obtain from our maven dependencies.

What this information is ultimately used for is we ship license texts of all of our transitive dependencies within the SDK binaries to integrate with the [oss-licenses-plugin](https://developers.google.com/android/guides/opensource).

As for what SDKs use which libraries, I'll have to double check and get back to you

**Comment by samtstern** [comment]
Thanks @vkryachko .

@lakersr8ers all of the code in this repository is licensed under the Apache 2.0 license here:
https://github.com/firebase/firebase-android-sdk/blob/master/LICENSE

As mentioned above, those other licenses are for our dependencies.  We have not modified those licenses in any way.  You'll have to make your own evaluation of whether having those transitive dependencies is acceptable to you.

**Comment by lakersr8ers** [comment]
That makes sense. However since you can pick and choose which pieces of firebase you want to integrate into your app (example being just having firebase-core and crashlytics), is it possible to omit those two transitive dependencies in any way? Like if I just include firebase-core and firebase-crashlytics will that transitive dependency exist?

**Comment by vkryachko** [comment]
@lakersr8ers some of the dependencies in question might just be false positives in our automation that determined these deps, I will take a look and respond here. Thanks for bringing this up.

**Comment by vkryachko** [comment]
@lakersr8ers sorry for the delay.

**tl;dr**: the licenses in question are not needed and I will remove them from firebase-android-sdk shortly.

The reason they are present now is because our open-sourcing automation picked up more licenses than it should have. Essentially the only things firebase SDKs depend on are present in their respective pom.xml files, so I will shortly create a PR that removes the licenses in question.
